lib/publiccloud/instance.pm

```diff
@@ -919,9 +919,18 @@ sub do_systemd_analyze_time {
 
 sub upload_supportconfig_log {
     my ($self, %args) = @_;
-    $self->ssh_script_run(cmd => 'sudo supportconfig -R /var/tmp -B supportconfig -x AUDIT', timeout => 600);
-    $self->ssh_script_run(cmd => 'sudo chmod 0644 /var/tmp/scc_supportconfig.txz');
-    $self->upload_log('/var/tmp/scc_supportconfig.txz', failok => 1, timeout => 600);
+    return unless (get_var('SUPPORTCONFIG_LOG', 1));
+    # to exclude nothings, set "-"
+    my $exclude = get_var('SUPPORTCONFIG_EXCLUDE', 'AUDIT');
+    $exclude = ($exclude =~ s/[-\s]+//gr) ? '-x ' . $exclude : '';
+    my $log = "/var/tmp/scc_supportconfig";
+    my $cmd = "timeout --preserve-status 900 sudo supportconfig -R " . dirname($log) . " -B supportconfig $exclude";
+    record_info('supportconfig start', "start time: " . localtime());
+    my $err = $self->ssh_script_run(cmd => "$cmd", timeout => 960, proceed_on_failure => 1);
+    $self->ssh_script_run(cmd => "ls -l " . dirname($log) . "; [ -f $log.txz ] || sudo tar -cv -f $log.txz -J $log; sudo chmod 0644 $log.txz", proceed_on_failure => 1);
+    $cmd = ($err) ? "Stopped at " . localtime() . ", Partial Log" : "Completed at " . localtime() . ", Log";
+    $self->upload_log("$log.txz", failok => 1, timeout => 900);
```

Comment:
Isn't 15 minutes too much?

Reply:
yes, may be, but in real cases I really don't have statistics on how much long it could last, in every feature in every platform, etc; i.e. on leap16 it lasted 3 min. 
But when an HANG condition occurrs, we shall give the time to the system to act/react, to be really sure that a stop is needed, executed by the timeout.
 
In origin it was 600sec, 10 min. Now set at 15 min. it is for a more safe range, that is bumped only in hanging cases.